web/api/query.go

```diff
@@ -48,6 +68,12 @@ func (serv MetricsService) Query(w http.ResponseWriter, r *http.Request) {
 	expr := params.Get("expr")
 	asText := params.Get("asText")
 
+	timestamp, err := parseTimestampOrNow(params.Get("timestamp"))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("invalid query timestamp %s", err), http.StatusBadRequest)
```

Comment:
The point is, if the client specifies "Accept: application/json" then it's not cool to send a different Content-Type. That's what I was saying with "return the error in the requested type".